src/ripple/rpc/impl/TransactionSign.cpp

```diff
@@ -1083,12 +1083,21 @@ Json::Value transactionSubmitMultiSigned (
             return RPC::make_error (rpcINVALID_PARAMS, err.str ());
         }
 
-        // The Fee field must be greater than zero.
-        if (stpTrans->getFieldAmount (sfFee) <= 0)
+        // The Fee field must be in XRP and greater than zero.
+        auto const fee = stpTrans->getFieldAmount (sfFee);
+
+        if (!isLegalNet (fee))
+        {
+            std::ostringstream err;
+            err << "Invalid " << sfFee.fieldName
+                << " field.  Fees must be specified in XRP.";
+            return RPC::make_error (rpcINVALID_PARAMS, err.str ());
+        }
+        if (fee <= 0)
```

Comment:
I remember this discussion, and I confirmed that the zero-fee regular key has to be signed by the master key only, so this is correct unless we intend to change that behavior.